### PR TITLE
Da/refactor 1

### PR DESCRIPTION
**Pull request by d-atkins**
## Description
Refactors repo test and `User#friend?` / `Follower#registered?` methods

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which maintains existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Notes
Removes `User#friend?` / `Follower#registered?` and replaces them with `User#friendable?` 

Adds edge case test back in to `user_can_see_github_repos_spec.rb`
  - headers added to webmock stubs
  - separate stubs are made for a second user's repos

## RSpec Results
```
56 examples, 0 failures

Coverage report generated for RSpec - 279 / 337 LOC (82.79%) covered.
```

### CODE REVIEW

File: spec/features/user/user_can_see_github_repos_spec.rb [hunk 1]
```diff
@@ -4,19 +4,29 @@
...
 
       allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(user)
-      stub_request(:get, "https://api.github.com/user/repos?direction=asc&sort=created")
-        .to_return(status: 200, body: repo_json)
-      stub_request(:get, "https://api.github.com/user/followers")
-        .to_return(status: 200, body: followers_json)
-      stub_request(:get, "https://api.github.com/user/following")
-        .to_return(status: 200, body: following_json)
+      requests = ['repos?direction=asc&sort=created', 'followers', 'following']
+      responses = [repo_json, followers_json, following_json]
```
Comment (KellyIB) [inline spec/features/user/user_can_see_github_repos_spec.rb:16]: wow.  love this refactor, David.  I had no idea you could do this!

**Review by KellyIB — APPROVED** [review]
